### PR TITLE
fix: Fall back on default SA for PipelineRuns

### LINKED ISSUE #7355: Pipeline/metapipeline SA can be accidentally undefined
I haven't figured out exactly how it's happening, but when you invoke `CreateQuickstartOptions` directly rather than via `jx create quickstart`, `CommonOptions.ServiceAccount` ends up as the empty string rather than defaulting to `tekton-bot`. We should make sure we never kick off a `PipelineRun` without an SA.

### PR DESCRIPTION
**Pull request by abayer**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

Also make it a constant so that we don't have "tekton-bot" scattered randomly through the codebase.

#### Which issue this PR fixes

fixes #7355

### CODE REVIEW

File: pkg/tekton/constants.go [hunk 1]
```diff
@@ -25,4 +25,7 @@ const (
 
 	// LabelType is the label added to Tekton CRDs for the type of pipeline.
 	LabelType = "jenkins.io/pipelineType"
+
+	// DefaultPipelineSA is the default service account used for pipelines
+	DefaultPipelineSA = "tekton-bot"
```
Comment (hferentschik) [inline pkg/tekton/constants.go:30]: 👍 - nice one

**Comment by abayer** [comment]
`tf-boot` is really hating me these days. =)

**Comment by hferentschik** [comment]
> tf-boot is really hating me these days. =)

also checking back the third time, since the PR still shows on my dash ;-)

**Pull request merged**